services/dynamic-sidecar/tests/unit/test_api_containers.py

```diff
@@ -411,56 +423,79 @@ def _expected_error_string(status_code: int) -> dict[str, Any]:
 async def test_outputs_watcher_disabling(
     test_client: TestClient,
     mock_event_filter_enqueue: AsyncMock,
+    caplog_info_debug: LogCaptureFixture,
 ):
     assert isinstance(test_client.application, FastAPI)
     outputs_context: OutputsContext = test_client.application.state.outputs_context
     outputs_manager: OutputsManager = test_client.application.state.outputs_manager
     outputs_manager.task_monitor_interval_s = WAIT_FOR_OUTPUTS_WATCHER / 10
-    WAIT_FOR_EVENTS = outputs_manager.task_monitor_interval_s * 10
+    WAIT_PORT_KEY_PROPAGATION = outputs_manager.task_monitor_interval_s * 10
+    CALLS_RECEIVED_BY_EVENT_FILTER = 3
 
-    async def _create_file_in_random_dir_in_inputs() -> int:
+    async def _create_port_key_events() -> int:
         random_subdir = f"{uuid4()}"
 
         await outputs_context.set_file_type_port_keys([random_subdir])
-        await asyncio.sleep(WAIT_FOR_EVENTS)
-
+        await asyncio.sleep(WAIT_PORT_KEY_PROPAGATION)
```

Comment:
Can't we have here a mock or similar that you can wait for instead of these sleeps? These sleeps make things slow or unreliable

Reply:
well then let's hope so..